application: serial_lte_modem: improve MQTT for long KeepAlive (#12441)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/MQTT_AT_commands.rst

::

   #XMQTTCON: <status>,<client_id>,<url>,<port>[,<sec_tag>],<keepalive>]

> * The ``<status>`` value is an integer.

Reviewer: ```suggestion
* The ``<status>`` value is an integer.
  It can accept one of the following values:
```

Reviewer: Just realized that this one is slightly wrong. Should be:
```suggestion
* The ``<status>`` value is an integer.
  It can have one of the following values:
```

>     * ``0`` - MQTT disconnected.
>     * ``1`` - MQTT connected.

Reviewer: ```suggestion
    * ``0`` - MQTT is not connected.
    * ``1`` - MQTT is connected.
```
This presumes that the read command is actively used to gain this information. If this is instead something that gets automatically reported when MQTT connects or disconnects, then the original, active wording is correct.

Author: The read command is just to report, not get input

Reviewer: What I was more looking for, is this command used by the user? Or is it something that gets automatically run?

Author: Yes it's for user to call
AT#XMQTTCON=....   SET comannd
AT#XMQTTCON?       READ command
AT#XMQTTCON=?    TEST command

::

>    #XMQTTCON: <status>,<client_id>,<url>,<port>[,<sec_tag>]

Author: Another round of doc review 🤦‍♂️

  It indicates the MQTT broker port.
* The ``<sec_tag>`` value is an integer.
  It indicates the credential of the security tag used for establishing a secure connection.
* The ``<keepalive>`` value is an integer.

>   It indicates the MQTT KeepAlive value (in seconds) used in current MQTT connection.

Reviewer: ```suggestion
  It indicates the MQTT Keep Alive value (in seconds) used in the current MQTT connection.
```
I believe the feature name is "Keep Alive" with two words.

Author: Yes

Examples
~~~~~~~~

::

   AT#XMQTTCON?
   #XMQTTCON: 1,"MyMQTT-Client-ID","","","mqtt.server.com",1883,60
   OK

Test command
------------

> The test command tests the existence of the command and provides information about the type of its subparameters.

Reviewer: ```suggestion
The test command tests the existence of the ``#XMQTTCON`` command and provides information about its subparameters.
```
Subparameters, or just parameters?

Author: Copy/paste from other SLM commands

Syntax
~~~~~~

::

   #XMQTTCON=?

Response syntax
~~~~~~~~~~~~~~~

::

>    #XMQTTCON: (0,1,2),<client_id>,<username>,<password>,<url>,<port>,<sec_tag>

Reviewer: These should be explained. If they're the same as for the Set command, you could use an `.. include::` like was recently added for the GNSS AT commands.

Author: The response should be
#XMQTTCON: (list of op),<client_id>,<username>,<password>,<url>,<port>,<sec_tag>

Examples
~~~~~~~~

::

   AT#XMQTTCON=?

>    #XMQTTCON: (0,1,2),<client_id>,<username>,<password>,<url>,<port>,<sec_tag>

Reviewer: This looks like the syntax, not an example response.
